[Wang Minhong] iP

ChrysanthemumT opened the pull request:
DukePro
DukePro Is the individual component of cs2103T, truly one of the mods of NUS.
> "I hope I find a Job" - me

Its
- text-based
- ~cool~

Features:
1.  Managing tasks
2.  Managing deadlines 
3.  Vim-motions (coming soon)

Here are some buttons
- [x] This pr is **cool**
- [ ] This pr is [lame](https://nus-cs2103-ay2324s2.github.io/website/schedule/week4/project.html)
- [ ] This pr is worse than cs2101 :-1:

If you are a Java programmer, what does this `main` function do?
``` java
public class Main {
    public static void main(String[] args) {
        int i = (byte) + (char) - (int) + (long) - 1;
        System.out.println(i);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

ChrysanthemumT committed:
basic skeleton

ChrysanthemumT committed:
added echo

ChrysanthemumT committed:
add, list

ChrysanthemumT committed:
Mark as Done complete

ChrysanthemumT committed:
level-4 added todo event deadline

ChrysanthemumT committed:
automated testing

ChrysanthemumT committed:
handled errors

ChrysanthemumT committed:
added delete

ChrysanthemumT committed:
Add saving feature

ChrysanthemumT committed:
Merge branch 'branch-Level-7'
Merge saving feature from branch to master

ChrysanthemumT committed:
Fix bug relating to the path of the saved file

ChrysanthemumT committed:
Add function to check ISO date

ChrysanthemumT committed:
Resolve merge conflict

ChrysanthemumT committed:
Add Date and Time to Tasks

ChrysanthemumT committed:
Merge branch 'branch-Level-8'

ChrysanthemumT committed:
Refactor project to include packages

ChrysanthemumT committed:
Refactor packages to subpackages of duke

ChrysanthemumT committed:
Merge remote-tracking branch 'origin/add-gradle-support'

ChrysanthemumT committed:
Add Junit Tests for Parser and Task

ChrysanthemumT committed:
Update savefile.txt

ChrysanthemumT committed:
Add HHmm time format to Parser

ChrysanthemumT committed:
Fix merge conflict occuring in savefile.txt

ChrysanthemumT committed:
Add JavaDoc to all Classes and methods

ChrysanthemumT committed:
Fix Style errors

ChrysanthemumT committed:
Merge branch 'branch-A-CodingStandard'

ChrysanthemumT committed:
Add a way to find keyword in tasklist

ChrysanthemumT committed:
Merge branch 'branch-Level-9'

Fidget-Spinner commented on src/main/java/duke/storage/Deadlines.java:
```diff
@@ -0,0 +1,65 @@
+    }
+
+    /**
+     * Returns a string representation of the deadline task, including its specific type identifier, the result of the superclass's toString method, and date-time details.
```
This line exceeds the coding style limit of 120 chars.

Fidget-Spinner commented on src/main/java/duke/storage/Events.java:
```diff
@@ -0,0 +1,91 @@
+
+
+    /**
+     * Returns a string representation of the event task, including its specific type identifier, the result of the superclass's toString method, and date-time details.
```
This line also exceeds 120 chars.

Fidget-Spinner commented on src/main/java/duke/storage/TaskList.java:
```diff
@@ -0,0 +1,132 @@
+/**
+ * The TaskList class represents a list of tasks in the Duke task manager.
+ * It manages operations such as adding, marking, unmarking, and removing tasks.
+ * The class also provides methods for obtaining the length of the task list and generating a formatted string representation of the tasks.
```
This line exceeds 120 chars.

Fidget-Spinner commented on src/main/java/duke/storage/Todos.java:
```diff
@@ -0,0 +1,27 @@
+    }
+
+    /**
+     * Returns a string representation of the todo task, including its specific type identifier and the result of the superclass's toString method.
```
This line also exceeds.

Fidget-Spinner reviewed (commented):
This is my first PR review for CS2103T where I will be reviewing coding standard issues.

Overally, looks good, just a few nits.

ChrysanthemumT committed:
Add checkstyle to gradle build

ChrysanthemumT committed:
Merge branch 'branch-A-CheckStyle'

ChrysanthemumT committed:
Add basic JavaFx GUI following the tutorial

ChrysanthemumT committed:
Style GUI for users experience

ChrysanthemumT committed:
Merge branch 'branch-Level-10'

ziiqii commented on src/main/java/duke/storage/Storage.java:
```diff
@@ -0,0 +1,63 @@
+     *
+     * @param taskList The TaskList object to be initialized with tasks from the save file.
+     */
+    public static void start(TaskList taskList) {
```
Maybe could have the method name be more descriptive, such as LoadTasks

ziiqii commented on src/main/java/duke/storage/Storage.java:
```diff
@@ -0,0 +1,63 @@
+     *
+     * @param input The input string representing the tasks to be saved.
+     */
+    public static void save(String input) {
```
Could also make it more descriptive, e.g. writeToSaveFile

ChrysanthemumT committed:
Fix Javadoc and stylechecks

ChrysanthemumT committed:
Merge branch 'branch-Level-10'

ziiqii commented on src/main/java/duke/storage/TaskList.java:
```diff
@@ -0,0 +1,133 @@
+     *
+     * @return The number of tasks in the task list.
+     */
+    public int taskLength() {
```
Maybe getTaskCount could be a more accurate method name?

ziiqii commented on src/main/java/duke/storage/TaskList.java:
```diff
@@ -0,0 +1,133 @@
+     *
+     * @param number The index of the task to be removed.
+     */
+    public String remove(int number) {
```
Maybe could be more specific e.g. removeAtIndex

ziiqii commented on src/main/java/duke/storage/TaskList.java:
```diff
@@ -0,0 +1,133 @@
+     * @param key The keyword to search for in task descriptions.
+     * @return A formatted string representation of tasks that match the search criteria.
+     */
+    public String find(String key) {
```
Maybe could be more descriptive e.g. findTasksByKeyword, but then again it might become too long haha

ziiqii reviewed (commented):
Overall, I find your code to be very neat and like it a lot!
The vast majority of your methods are also well named, LGTM!

ChrysanthemumT committed:
Add assertions to check parser output

ChrysanthemumT committed:
Merge pull request #1 from ChrysanthemumT/branch-A-Assertions

Branch-A-Assertions

ChrysanthemumT committed:
Refactor Parser Class
Logic in switch statement mostly extracted out. Change handling of Datetime
to entirely use LocalDateTime instead of manually parsing. Event and deadline
classes now utilise a String date time, as parser converts DateTimes to String.

Fix javadocs for old methods, and add new ones for extracted methods. Fix some styling
issues to follow course standard as well as coding practices. Mostly fixing guard clause.

ChrysanthemumT committed:
Merge pull request #2 from ChrysanthemumT/branch-A-CodeQuality

Refactor Parser Class

ChrysanthemumT committed:
Add C-Priority

Now possible to tag a task with priority, either high or low.
new task added are set with no priority, can now be set using
prioriy {index} {priority}

Keeping the same format as marking an unmarking

ChrysanthemumT committed:
Merge pull request #3 from ChrysanthemumT/branch-BCD-Extension

Add C-Priority

ChrysanthemumT committed:
Fix style issues with long methods

ChrysanthemumT committed:
Add User guide in README

ChrysanthemumT committed:
Add more details to user guide

ChrysanthemumT committed:
Fix savefile by changing to relative path

ChrysanthemumT committed:
Fix checkstyle error
